src/Orleans/Messaging/IncomingMessageBuffer.cs

```diff
@@ -128,6 +131,19 @@ public bool TryDecodeMessage(out Message msg)
             headerLength = 0;
             bodyLength = 0;
 
+            return true;
```

Comment:
Regarding buffer leaks in general, we have a best effort method of buffer management, as it's better to leak buffers than to double free.  With that in mind, leaking some buffers in rare cases, while not desirable, is not harmful.  The only case where these buffers would be 'leaked' is in the case where this entire buffer is released, and this change has not introduced nor significantly aggravated that issue.

As for the behavior here, we never really release the buffer, we just shrink it some times if needed.  Instead of adjusting it every message we pull out, we are now adjusting it once prior to the next socket read.  We always expect a read, even if there is no data to be read.  If there is no next socket read, we're either shutting down, or this buffer has been thrown away (pre-existing buffer leak).